### PR TITLE
Updated setup script to publish single binary file

### PR DESCRIPTION
**Pull request by kuqin12**
# Preface

Please ensure you have read the [contribution docs](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md) prior
to submitting the pull request. In particular,
[pull request guidelines](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md#pull-request-best-practices).

## Description

This change switched from the original consumed cx_freeze to pyinstaller to support the single binary publication process. The consumers are expected to collect all potential configuration definition files to one folder and pass to the `setup.py` script.

This script will embed all potentially needed modules from ConfigEditor.py and the optionally supplied XML files into one exe file output.

In the meantime, minor changes are added to not validate xml schema when the script is running in frozen mode to avoid dependency overhead. The local developing environment will run the schema files as usual.

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [x] Impacts functionality?
- [ ] Impacts security?
- [x] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?

## How This Was Tested

This is manually tested. For Windows output, the executable works as expected, but for Linux builds, there is hard requirements on GLIBC versions (building environment needs to be <= running environment).

## Integration Instructions

For automations that already uses this setup.py with cx_freeze, the developer needs to update the pip module dependency to pyinstaller and invoke the script differently:

- `setup.py [--ConfFolderPath CONFFOLDERPATH] [--OutputFilePath OUTPUTFILEPATH]`

### CODE REVIEW

**Review by os-d — APPROVED** [review]

File: SetupDataPkg/Tools/ConfigEditor.py [hunk 1]
```diff
@@ -535,6 +535,21 @@ def __init__(self, master=None):
...
+            # The application is frozen, pre-populate collected definition files
+            print("Running bundled ConfigEditor! Load pre-populated definition files.\n")
+
+            bundle_dir = sys._MEIPASS
+
+            # The collected definitions will be put under "ConfDefinitions" folder in the bundle directory
+            for subdir, _, files in os.walk(os.path.join(bundle_dir, "ConfDefinitions")):
+                for file in files:
+                    sub_path = os.path.join(subdir, file)
+                    print (sub_path)
```
Comment (apop5) [inline SetupDataPkg/Tools/ConfigEditor.py:548]: Is the print intended? or is it left over from debugging?
```suggestion

```

File: SetupDataPkg/Tools/ConfigEditor.py [hunk 2]
```diff
@@ -535,6 +535,21 @@ def __init__(self, master=None):
...
+                    # if sub_path.endswith(".xml") or sub_path.endswith(".yaml") or sub_path.endswith(".yml"):
```
Comment (apop5) [inline SetupDataPkg/Tools/ConfigEditor.py:549]: do you need the commented out code?

**Review by apop5 — APPROVED** [review]

File: SetupDataPkg/Tools/ConfigEditor.py (hunk 1 above)
Comment (kuqin12) [reply SetupDataPkg/Tools/ConfigEditor.py:547]: Ah, good call, it should not be here. Will update.

Comment (kuqin12) [reply SetupDataPkg/Tools/ConfigEditor.py]: Removed.

File: SetupDataPkg/Tools/ConfigEditor.py (hunk 2 above)
Comment (kuqin12) [reply SetupDataPkg/Tools/ConfigEditor.py]: Thanks, fixed.

**Pull request merged**